docker-compose.yml

```diff
@@ -0,0 +1,68 @@
+version: '3.7'
+services:
+  replication-solr:
+    image: solr:8.1.1
+    ports:
+     - "8983:8983"
+    volumes:
+      - replication-data:/var/solr
+    networks:
+      - replication
+    configs:
+      - source: replication-solr-schema
+        target: /opt/solr/server/solr/configsets/_default/conf/managed-schema
+    entrypoint:
+      - bash
+      - "-c"
+      - "precreate-core replication_item; precreate-core replication_site; precreate-core replication_status; precreate-core replication_config; exec solr -f"
+  ion-replication:
+    image: cnxta/replication:0.3.0-SNAPSHOT
+    networks:
+      - replication
+    entrypoint:
+      - bash
+      - "-c"
+      - "java \
+        -Dssl.system.properties=/opt/replication/ssl.properties \
+        -Djavax.net.ssl.keyStore=/opt/replication/serverKeystore.jks
+        -Djavax.net.ssl.trustStore=/opt/replication/serverTruststore.jks
+        -Duser.dir=/opt/replication/ \
+        -jar /opt/replication/app.jar --spring.config.location=file:/opt/replication/"
+    secrets:
+      - source: replication-keystore
+        target: /opt/replication/serverKeystore.jks
+      - source: replication-truststore
+        target: /opt/replication/serverTruststore.jks
+      - source: replication-ssl
+        target: /opt/replication/ssl.properties
+    configs:
+      - source: replication-spring-config
+        target: /opt/replication/application.properties
+      - source: replication-jobs
+        target: /opt/replication/config/configs.json
+      - source: replication-sites
+        target: /opt/replication/config/sites.json
+configs:
+  replication-spring-config:
+    external: true
+  replication-jobs:
+    external: true
+  replication-sites:
+    external: true
+  replication-solr-schema:
+    external: true
+secrets:
+  replication-keystore:
+    external: true
+  replication-truststore:
+    external: true
+  replication-ssl:
+    external: true
+volumes:
+  replication-data:
+    external: true
+    name: replication-data
+networks:
+  replication:
+    external: true
+    name: replication
```

Comment:
❓ could you maybe explain this one as well? It looks like these are externally set properties but how do they get set?

Reply:
You don't need to specify the name if it matches already.

Reply:
removed name and external